src/Internal/Payment/State/InitialState.php

```diff
@@ -109,6 +121,13 @@ public function start_processing( PaymentRequest $request ) {
 		// Start multiple verification checks.
 		$this->process_order_phone_number();
 
+		if ( $this->fraud_prevention_service->is_enabled()
+			&& ! $this->fraud_prevention_service->verify_token( $this->get_context()->get_fraud_prevention_token() ) ) {
```

Comment:
This is a pet peeve of mine, but could you please integrate `is_enabled` within `verify_token` of the service? I don't see why external callers would need to check both. Then this would be a bit cleaner:



```suggestion
		if ( ! $this->fraud_prevention_service->verify_token( $this->get_context()->get_fraud_prevention_token() ) ) {
```

Reply:
yeah, that's something a bit strange under my eyes too. Will update it.